vm.go

```diff
@@ -102,13 +102,19 @@ func (l *State) equalObjects(t1, t2 value) bool {
 	case *userData:
 		if t1 == t2 {
 			return true
+		} else if t2, ok := t2.(*userData); ok && t2 != nil {
+			tm = l.equalTagMethod(t1.metaTable, t2.metaTable, tmEq)
+		} else if !ok || t2 == nil {
+			return false
```

Comment:
That actually fails tests, so I'm inclined to leave it as-is to avoid breaking anything there.